database/db.go

```diff
@@ -0,0 +1,139 @@
+package database
+
+import (
+	"database/sql"
+
+	_ "github.com/mattn/go-sqlite3" // Import the SQLite driver
+)
+
+// TxStatus represents the different states of a transaction.
+type TxStatus int
+
+// Transaction status constants
+const (
+	StatusPending TxStatus = iota
+	StatusBroadcasted
+	StatusConfirmed
+)
+
+// SQL queries
+const (
+	insertTransactionSQL       = "INSERT INTO transactions(txid, rawtx, status) values(?,?,?)"
+	getPendingTransactionsSQL  = "SELECT txid, rawtx, status FROM transactions WHERE status = ?"
+	getTransactionByTxidSQL    = "SELECT txid, rawtx, status FROM transactions WHERE txid = ?"
+	updateTransactionStatusSQL = "UPDATE transactions SET status = ? WHERE txid = ?"
+	createTransactionsTableSQL = `
+        CREATE TABLE IF NOT EXISTS transactions (
+            txid TEXT PRIMARY KEY,
+            rawtx TEXT NOT NULL,
+            status INTEGER NOT NULL NOT NULL
+        )
+    `
+)
+
+// Tx represents a transaction record in the database.
+type Tx struct {
+	BtcTxID uint64   `json:"txid"`
+	RawTx   string   `json:"rawtx"`
+	Status  TxStatus `json:"status"`
+	// TODO: other fields
+}
+
+// DB holds the database connection and provides methods for interacting with it.
+type DB struct {
+	conn *sql.DB
+}
+
+// NewDB creates a new DB instance and initializes the database connection.
+func NewDB(dbPath string) (*DB, error) {
+	db := &DB{} // Create a new DB instance
+
+	// Initialize the database connection
+	var err error
+	db.conn, err = sql.Open("sqlite3", dbPath)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = db.conn.Exec(createTransactionsTableSQL)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Let's move DB init outside of DB creation.

Reply:
Done